migration-guide.md

```diff
@@ -0,0 +1,140 @@
+# Migration guide
+
+This migration guide details how to migrate to `package:platform` version 3.2,
+from earlier Dart platform APIs located in:
+
+  - `dart:io`
+  - `package:platform` version 3.1
+  - `package:os_detect`
+
+# From `dart:io`
+
+The `dart:io` core library is only available on Dart native platforms 
+(and, for example, not available on Dart web platforms). 
+We recommend migration to `package:platform` version 3.2, as the APIs in
+`package:platform/platform.dart` are available on all current platforms.
+
+_Note: This migration can be performed automatically by running `dart fix`._
+
+### General platform APIs
+
+The `dart:io` library exposes host platform information as static members of the
+`Platform` class. This API has historically been made available on some
+platforms that don't otherwise support `dart:io`.
+
+To migrate uses of that API to `package:platform` version 3.2, use the
+similarly-named members on the `Platform.current` object.
+
+Migrate from:
+```dart
+import 'dart:io';
+
+bool onAndroid = Platform.isAndroid;
+```
+
+To:
+```dart
+import 'package:platform/platform.dart';
+
+bool onAndroid = Platform.current.isAndroid;
+```
+
+### Native-only 
+
+The remaining `dart:io` APIs on the `Platform` class, are only available on
+native platforms (and not, for example, on the web).
+
+To migrate those to `package:platform`, use the `nativePlatform` getter. Note
+that this will return `null` when not on a native platform. You can test if you
+are on a native platform using `Platform.current.isNative`.
+
+Migrate from:
+```dart
+import 'dart:io';
+
+String hostname = Platform.localHostname;
+```
+
+To:
+```dart
+import 'package:platform/platform.dart';
+
+if (Platform.current.isNative) {
+  String hostname = Platform.current.nativePlatform!.localHostname;
+}
```

Comment:
Or use a pattern, to avoid the double check (first `.isNative` then `!` too):
```dart
if (Platform.current.nativePlatform case var nativePlatform?) {
   String hostname = nativePlatofrm.localHostname;
}
```

Reply:
I think that lots of developers think this is very hard to read...